apps/web/src/services/cost-explorer/CostExplorerLNB.vue

```diff
@@ -167,7 +183,22 @@ onUnmounted(() => {
     costQuerySetStore.$reset();
 });
 
-costQuerySetStore.listCostQuerySets();
+
+watch(() => route.params, (params) => {
+    if (params.dataSourceId && params.costQuerySetId) {
+        costQuerySetStore.$patch({
+            selectedDataSourceId: params.dataSourceId,
+            selectedQuerySetId: params.costQuerySetId,
+        });
+        costQuerySetStore.listCostQuerySets();
+    }
```

Comment:
question: What happens if costQuerySetId does not exist?

Reply:
This will be handled by `beforeEnter` in `route.ts`. Don't this look safe to you?